core/src/main/scala/org/apache/spark/storage/BlockManager.scala

```diff
@@ -242,8 +244,7 @@ private[spark] class BlockManager(
 
   private var blockReplicationPolicy: BlockReplicationPolicy = _
 
-  private var blockManagerDecommissioning: Boolean = false
-  private var decommissionManager: Option[BlockManagerDecommissionManager] = None
+  @volatile private var decommissioner: Option[BlockManagerDecommissioner] = None
```

Comment:
If we receive two decommissioning messages at the same time. we don't want to launch two of them.

Reply:
Hmm, the creation to `decommissioner` is guarded by a lock.

Reply:
That's true. If we drop it we might also accept remove block puts after we've started decommissioning though. Depends on how much we want to avoid that.

Reply:
I think I'm going to leave it volatile for now, I'd like to avoid remote block puts once we're in decommissioning because we depend on not getting new blocks except from tasks to figure out when it is safe to exit.